pkg/common-controller/snapshot_controller.go

```diff
@@ -943,6 +943,18 @@ func (ctrl *csiSnapshotCommonController) updateSnapshotStatus(snapshot *crdv1.Vo
 	if content.Status != nil && content.Status.ReadyToUse != nil {
 		readyToUse = *content.Status.ReadyToUse
 	}
+	var volumeSnapshotErr *crdv1.VolumeSnapshotError
+	if content.Status != nil && content.Status.Error != nil {
+		volumeSnapshotErr = &crdv1.VolumeSnapshotError{}
+		if content.Status.Error.Time != nil {
+			errTime := *content.Status.Error.Time
+			volumeSnapshotErr.Time = &errTime
+		}
+		if content.Status.Error.Message != nil {
+			msg := *content.Status.Error.Message
+			volumeSnapshotErr.Message = &msg
```

Comment:
@xing-yang Dont we have to record this event once we set it on the VolumeSnapshot object?